core/src/gateway_ll.c

```diff
@@ -85,19 +93,30 @@ GATEWAY_HANDLE Gateway_LL_Create(const GATEWAY_PROPERTIES* properties)
 						if (module == NULL)
 						{
 							LogError("Gateway_LL_Create(): Unable to add module '%s'. The gateway will be destroyed.", entry->module_name);
-							while (gateway->modules != NULL && VECTOR_size(gateway->modules) > 0)
-							{
-								MODULE_DATA* module_data = (MODULE_DATA*)VECTOR_front(gateway->modules);
-								//By design, there will be no NULL module_data pointers in the vector
-								gateway_removemodule_internal(gateway, module_data);
-							}
-							VECTOR_destroy(gateway->modules);
-							MessageBus_Destroy(gateway->bus);
-							free(gateway);
+							gateway_destroy_internal(gateway);
 							gateway = NULL;
 						}
 					}
 				}
+				if (gateway != NULL)
+				{
+					/* TODO: Seperate the gateway init from gateway start-up so that plugins have the chance
+					 * register themselves */
+					/*Codes_SRS_GATEWAY_LL_26_001: [ This function shall initialize attached Gateway Events callback system and report GATEWAY_CREATED event. ] */
+					gateway->event_system = EventSystem_Init();
+					/*Codes_SRS_GATEWAY_LL_26_002: [ If Gateway Events module fails to be initialized the gateway module shall be destroyed with no events reported. ] */
+					if (gateway->event_system == NULL)
+					{
+						LogError("Gateway_LL_Create(): Unable to initialize callback system");
+						gateway_destroy_internal(gateway);
+						gateway = NULL;
+					}
+					else
+					{
+						/*Codes_SRS_GATEWAY_LL_26_001: [ This function shall initialize attached Gateway Events callback system and report GATEWAY_CREATED event. ] */
+						EventSystem_ReportEvent(gateway->event_system, gateway, GATEWAY_CREATED);
```

Comment:
@avranju Not right now, there's a team TODO task to separate Gateway init from starting up so that the callbacks can be registered.
This report is future proofing for now to ensure everything is logged correctly (also for potential side-effects, such as internal callbacks).